app/lib/jsonvalidate.ts

```diff
@@ -0,0 +1,93 @@
+var fs = require('fs');
+var shell = require('shelljs');
+var check = require('validator');
+
+export class JsonValidate {
+	/*
+	 * Checks a json file for correct formatting against some validation function
+	 * @param jsonFilePath path to the json file being validated
+	 * @param jsonValidationFunction function that validates parsed json data against some criteria
+	 * @return the parsed json file
+	 * @throws InvalidDirectoryException if json file doesn't exist, InvalidJsonException on failed parse or *first* invalid field in json
+	*/
+	public validateJson(jsonFilePath: string, jsonValidationFunction: (path: string, jsonData: any) => void, jsonMissingErrorMessage?: string): any {
+		var jsonMissingErrorMsg: string = jsonMissingErrorMessage || 'specified json file does not exist.'
+        this.exists(jsonFilePath, jsonMissingErrorMsg);
+
+        var taskJson;
+        try {
+            taskJson = JSON.parse(fs.readFileSync(jsonFilePath, 'utf8'));
+        }
+        catch (jsonError) {
+            throw new InvalidJsonException("Invalid task json: " + jsonError);
```

Comment:
Let's make this function return a Q.Promise<void>.  An error would reject with a good message